### DISCUSSION TITLE
经过客户端挂机code45退回到了235但是我无法使用修复插件

### DISCUSSION THREAD

**Discussion by AlphaNight001** [simple-robot/Discussions#136] [Q&A]
### 涉及的编程语言

Java

### 涉及的组件库

mirai (QQ)

### 疑问描述

根据文档说的方法我已经退回到了235，但是插件我貌似错误引用了
fix-protocol-version

我将这个jar包在IDEA的springboot项目中作为jar包引入了

但是我不太会使用，另外官方说法是：下载 mirai2.jar 放到 plugins ，重启 Mirai Console 即可

Plugins这个目录是maven的目录吗？

而且我貌似并没有Mirai Console这个插件  以下是我的maven代码
<?xml version="1.0" encoding="UTF-8"?>
<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
    <modelVersion>4.0.0</modelVersion>
    <parent>
        <groupId>org.springframework.boot</groupId>
        <artifactId>spring-boot-starter-parent</artifactId>
        <version>3.0.0</version>
        <relativePath/> 
    </parent>
    <groupId>com.example.Alpha</groupId>
    <artifactId>QQMessage</artifactId>
    <version>0.0.1-SNAPSHOT</version>
    <packaging>jar</packaging>
    <name>QQMessage</name>
    <description>QQMessage</description>
    <properties>
        <java.version>17</java.version>
    </properties>
    <dependencies>

        <dependency>
            <groupId>org.springframework.boot</groupId>
            <artifactId>spring-boot-starter-webflux</artifactId>
        </dependency>

            
            <dependency>
                <groupId>org.jsoup</groupId>
                <artifactId>jsoup</artifactId>
                <version>1.14.1</version>
            </dependency>

        <dependency>
            <groupId>com.alibaba</groupId>
            <artifactId>fastjson</artifactId>
            <version>1.2.76</version>
        </dependency>

        <dependency>
            <groupId>org.springframework.boot</groupId>
            <artifactId>spring-boot-starter-jdbc</artifactId>
        </dependency>
        <dependency>
            <groupId>org.springframework.boot</groupId>
            <artifactId>spring-boot-starter-web</artifactId>
            <exclusions>
                <exclusion>
                    <groupId>org.springframework.boot</groupId>
                    <artifactId>spring-boot-starter-tomcat</artifactId>
                </exclusion>
            </exclusions>
            <scope>provided</scope>
        </dependency>
        <dependency>
            <groupId>org.springframework.boot</groupId>
            <artifactId>spring-boot-starter-websocket</artifactId>
        </dependency>
        <dependency>
            <groupId>org.mybatis.spring.boot</groupId>
            <artifactId>mybatis-spring-boot-starter</artifactId>
            <version>3.0.0</version>
        </dependency>
        
        <dependency>
            <groupId>love.forte.simbot.boot</groupId>
            <artifactId>simboot-core-spring-boot-starter</artifactId>
            
            <version>3.0.0</version>
        </dependency>
        <dependency>
            <groupId>love.forte.simbot.component</groupId>
            <artifactId>simbot-component-mirai-core</artifactId>
            <version>3.0.0.0-RC.2</version>
        </dependency>
        <dependency>
            <groupId>com.mysql</groupId>
            <artifactId>mysql-connector-j</artifactId>
            <scope>runtime</scope>
        </dependency>
        <dependency>
            <groupId>org.springframework.boot</groupId>
            <artifactId>spring-boot-starter-tomcat</artifactId>
            <scope>provided</scope>
        </dependency>
        <dependency>
            <groupId>org.springframework.boot</groupId>
            <artifactId>spring-boot-starter-test</artifactId>
            <scope>test</scope>
        </dependency>
        <dependency>
            <groupId>org.jetbrains.kotlinx</groupId>
            <artifactId>kotlinx-serialization-json</artifactId>
            <version>1.4.0</version>
        </dependency>
        <dependency>
            <groupId>org.springframework.boot</groupId>
            <artifactId>spring-boot-autoconfigure</artifactId>
            <version>3.0.0</version>
        </dependency>
    </dependencies>

    <build>
        <plugins>
            <plugin>
                <groupId>org.springframework.boot</groupId>
                <artifactId>spring-boot-maven-plugin</artifactId>
            </plugin>
        </plugins>
    </build>

</project>

**Comment by AlphaNight001** [comment]
Mirai Console 我找到了，是否是需要在Springboot启动类中直接调用FixProtocolVersion.update();?

**Comment by ForteScarlet** [comment]
> 我将这个jar包在IDEA的springboot项目中作为jar包引入了

引入后，在main方法开头使用 `FixProtocolVersion.update()` 即可。

**Reply by AlphaNight001** [reply]
非常感谢，现在code从45变成235插件加载之后又变成237  我现在在尝试解决237问题，真的快奔溃了┭┮﹏┭┮

**Reply by ForteScarlet** [reply]
协议类型我记得要设置成Pad类的，比如安卓pad

**Reply by AlphaNight001** [reply]
好的我试试
